src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.cs

```diff
@@ -84,7 +84,8 @@ private static void OnCompilationStart(CompilationStartAnalysisContext context)
                     return;
                 }
 
-                operationContext.ReportDiagnostic(Diagnostic.Create(Rule, invocation.Syntax.GetLocation()));
+                var properties = new Dictionary<string, string> { ["method"] = invocation.TargetMethod.Name }.ToImmutableDictionary();
```

Comment:
Can you define a constant string field to hold the key and use the field both here and in the code fixer?